src/groupeddataframe/fastaggregates.jl

```diff
@@ -122,10 +122,10 @@ for (op, initf) in ((:max, :typemin), (:min, :typemax))
             # !ismissing check is purely an optimization to avoid a copy later
             outcol = similar(incol, condf === !ismissing ? S : T, length(gd))
             # Comparison is possible only between CatValues from the same pool
-            if incol isa CategoricalVector
-                U = Union{CategoricalArrays.leveltype(outcol),
-                          eltype(outcol) >: Missing ? Missing : Union{}}
-                outcol = CategoricalArray{U, 1}(outcol.refs, incol.pool)
+            incolT = typeof(incol).name
+            if incolT.name === :CategoricalArray &&
+                nameof(incolT.module) === :CategoricalArrays
+                outcol = Vector{eltype(incol)}(undef, length(gd))
```

Comment:
Another ugly option that might work (and should be faster):
```julia
    outcol.pool = incol.pool
```
(might be clearer to change the `if` to check `outcol` rather than `incol` even if that's equivalent)

Reply:
OK - I will try this.